types/types.go

```diff
@@ -43,14 +47,22 @@ func NewWarpBlockInfo(header *types.Header, ethClient ethclient.Client) (*WarpBl
 	)
 	// Check if the block contains warp logs, and fetch them from the client if it does
 	if header.Bloom.Test(WarpPrecompileLogFilter[:]) {
-		logs, err = ethClient.FilterLogs(context.Background(), interfaces.FilterQuery{
-			Topics:    [][]common.Hash{{WarpPrecompileLogFilter}},
-			Addresses: []common.Address{warp.ContractAddress},
-			FromBlock: big.NewInt(int64(header.Number.Uint64())),
-			ToBlock:   big.NewInt(int64(header.Number.Uint64())),
-		})
-		if err != nil {
-			return nil, err
+		// The node serving the filter logs request may be behind the node serving the block header request,
```

Comment:
Would it be worth moving this out to a helper? I could see use tinkering with it in the future to adjust the delay/number of retries, etc.

Reply:
Good call, added a helper